test/types/decorator.py

```diff
@@ -0,0 +1,31 @@
+from dataclasses import dataclass
+
+import tensorflow as tf
```

Comment:
Tensorflow is quite a heavy dependency, that sometimes get skipped in some test environments. Can we switch this out with just numpy?